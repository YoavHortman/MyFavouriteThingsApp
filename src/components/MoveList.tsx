import * as React from "react";
import {FavouriteMovie} from "../models/movie";
import {ListItem} from './ListItem';

export interface MovieListProps {
    movies: FavouriteMovie[];
    onRateClick: (id: number, rating: number) => void;
}

export class MoveList extends React.Component<MovieListProps> {
    render() {
        return (
            <div>
                {this.props.movies.map(movie => {
                    return (
                        <ListItem
                            key={movie.id}
                            item={movie}
                            onRateClick={this.props.onRateClick}
                        />
                    )
                })}
            </div>
        )
    }
}

