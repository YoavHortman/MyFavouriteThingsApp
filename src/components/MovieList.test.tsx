import * as React from "react";
import * as renderer from "react-test-renderer";
import {MoveList} from "./MoveList";

it('Should render empty list', () => {
    const tree = renderer
        .create(
            <MoveList
                movies={[]}
                onRateClick={() => {
                }}
            />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Should render a list with 2 items', () => {
    const tree = renderer
        .create(
            <MoveList
                movies={[
                    {
                        id: 10,
                        name: "I like lots of movies9",
                        createdBy: "me",
                        lengthInMinutes: 100,
                        rating: 5
                    }]}
                onRateClick={() => {
                }}
            />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
