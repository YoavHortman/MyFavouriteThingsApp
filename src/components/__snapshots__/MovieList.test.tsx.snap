// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a list with 2 items 1`] = `
<div>
  <div
    className="ListItem_root"
  >
    <div
      className="ListItem_rowContainer"
    >
      <span
        className="ListItem_rowTitle"
      >
        Name:
      </span>
      I like lots of movies9
    </div>
    <div
      className="ListItem_rowContainer"
    >
      <span
        className="ListItem_rowTitle"
      >
        Created by:
      </span>
      me
    </div>
    <div
      className="ListItem_rowContainer"
    >
      <span
        className="ListItem_rowTitle"
      >
        Rating:
      </span>
      <div
        className="dv-star-rating"
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <input
          checked={true}
          className="dv-star-rating-input"
          id="Rating_5"
          name="Rating"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
              "marginLeft": -9999,
              "position": "absolute",
            }
          }
          type="radio"
          value={5}
        />
        <label
          className="dv-star-rating-star dv-star-rating-full-star"
          htmlFor="Rating_5"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "color": "#ffb400",
              "cursor": "pointer",
              "float": "right",
            }
          }
        >
          <i
            style={
              Object {
                "fontStyle": "normal",
              }
            }
          >
            ★
          </i>
        </label>
        <input
          checked={false}
          className="dv-star-rating-input"
          id="Rating_4"
          name="Rating"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
              "marginLeft": -9999,
              "position": "absolute",
            }
          }
          type="radio"
          value={4}
        />
        <label
          className="dv-star-rating-star dv-star-rating-full-star"
          htmlFor="Rating_4"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "color": "#ffb400",
              "cursor": "pointer",
              "float": "right",
            }
          }
        >
          <i
            style={
              Object {
                "fontStyle": "normal",
              }
            }
          >
            ★
          </i>
        </label>
        <input
          checked={false}
          className="dv-star-rating-input"
          id="Rating_3"
          name="Rating"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
              "marginLeft": -9999,
              "position": "absolute",
            }
          }
          type="radio"
          value={3}
        />
        <label
          className="dv-star-rating-star dv-star-rating-full-star"
          htmlFor="Rating_3"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "color": "#ffb400",
              "cursor": "pointer",
              "float": "right",
            }
          }
        >
          <i
            style={
              Object {
                "fontStyle": "normal",
              }
            }
          >
            ★
          </i>
        </label>
        <input
          checked={false}
          className="dv-star-rating-input"
          id="Rating_2"
          name="Rating"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
              "marginLeft": -9999,
              "position": "absolute",
            }
          }
          type="radio"
          value={2}
        />
        <label
          className="dv-star-rating-star dv-star-rating-full-star"
          htmlFor="Rating_2"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "color": "#ffb400",
              "cursor": "pointer",
              "float": "right",
            }
          }
        >
          <i
            style={
              Object {
                "fontStyle": "normal",
              }
            }
          >
            ★
          </i>
        </label>
        <input
          checked={false}
          className="dv-star-rating-input"
          id="Rating_1"
          name="Rating"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
              "marginLeft": -9999,
              "position": "absolute",
            }
          }
          type="radio"
          value={1}
        />
        <label
          className="dv-star-rating-star dv-star-rating-full-star"
          htmlFor="Rating_1"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "color": "#ffb400",
              "cursor": "pointer",
              "float": "right",
            }
          }
        >
          <i
            style={
              Object {
                "fontStyle": "normal",
              }
            }
          >
            ★
          </i>
        </label>
      </div>
    </div>
    <div
      className="ListItem_rowContainer"
    >
      <span
        className="ListItem_rowTitle"
      >
        Duration:
      </span>
      100 minutes
    </div>
  </div>
</div>
`;

exports[`Should render empty list 1`] = `<div />`;
